test/small_alloc_create.c

```diff
@@ -0,0 +1,39 @@
+#include <small/small.h>
+#include <small/quota.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <time.h>
+#include "unit.h"
+
+#undef slab_cache_sysconf
+long
+slab_cache_sysconf(int name)
+{
+	switch (name) {
+	case _SC_PAGESIZE:
+		return 16384;
+	default:;
+	}
+	return sysconf(name);
+}
+
+int main()
+{
+	struct quota quota;
+	struct slab_arena arena;
+	struct slab_cache cache;
+	struct small_alloc alloc;
+	float actual_alloc_factor;
+
+	header();
+	quota_init(&quota, UINT_MAX);
+	slab_arena_create(&arena, &quota, 0, 4000000, MAP_PRIVATE);
+	slab_cache_create(&cache, &arena);
+	small_alloc_create(&alloc, &cache, 16,
+			   sizeof(intptr_t), 1.05,
+			   &actual_alloc_factor);
+	small_alloc_destroy(&alloc);
+	slab_cache_destroy(&cache);
+	slab_arena_destroy(&arena);
+	footer();
+}
```

Comment:
I talked to @alyapunov and here's what we think:

1. Weak linking looks dubious - the default implementation is weak, i.e. it can be replaced (maybe accidentally!) by a program that links this library, which makes no sense.
2. The test should be more extensive. Ideally we should compile the existing `test/small_alloc.c` for different page sizes (4K, 8K, 16K, 32K, 64K) and run them all.

Let's define `SLAB_MIN_ORDER0_SIZE` macro in CMake. By default it should be set to `sysconf(_SC_PAGESIZE)`. Let's use this configuration option to compile `test/small_alloc.c` with different page sizes.

Reply:
Fixed